components/Modal/index.d.ts

```diff
@@ -27,14 +24,14 @@ export interface ModalProps {
   };
 }
 
-export default class Modal extends Component<ModalProps> {
-  static Header: ModalHeader;
-
-  static Title: Title;
-
-  static HeaderText: HeaderText;
+interface CompoundedModal extends React.FunctionComponent<ModalProps> {
+  Header: React.FunctionComponent<ModalProps>;
+  Title: React.FunctionComponent<ModalProps>;
+  HeaderText: React.FunctionComponent<ModalProps>;
+  Content: React.FunctionComponent<ModalProps>;
+  Footer: React.FunctionComponent<ModalProps>;
+}
 
-  static Content: ModalContent;
+const Modal: CompoundedModal = () => null;
```

Comment:
```suggestion
```
Don't constantly cry on type file

Reply:
this treatment was necessary to "convert" the class component into a functional one